compiler/rustc_codegen_llvm/src/consts.rs

```diff
@@ -165,19 +165,9 @@ fn check_and_apply_linkage<'ll, 'tcx>(
     if let Some(linkage) = attrs.linkage {
         debug!("get_static: sym={} linkage={:?}", sym, linkage);
 
-        // If this is a static with a linkage specified, then we need to handle
-        // it a little specially. The typesystem prevents things like &T and
-        // extern "C" fn() from being non-null, so we can't just declare a
-        // static and call it a day. Some linkages (like weak) will make it such
-        // that the static actually has a null value.
-        let llty2 = if let ty::RawPtr(ref mt) = ty.kind() {
-            cx.layout_of(mt.ty).llvm_type(cx)
-        } else {
-            cx.sess().emit_fatal(LinkageConstOrMutType { span: cx.tcx.def_span(def_id) })
-        };
         unsafe {
             // Declare a symbol `foo` with the desired linkage.
-            let g1 = cx.declare_global(sym, llty2);
+            let g1 = cx.declare_global(sym, cx.type_i8());
```

Comment:
Same here.

Reply:
The actual type here doesn't matter because this is a declaration.

I was previously under the impression that Rust's LLVM uses opaque pointers so I didn't bother with a pointer cast here. After creating the PR I discovered that Rust apparently still supports old versions of LLVM that use non-opaque pointers, so I will adjust the code below this to add one.

Reply:
Done.